src/main/java/com/zufar/onlinestore/payment/service/PaymentEventParser.java

```diff
@@ -0,0 +1,27 @@
+package com.zufar.onlinestore.payment.service;
+
+import com.stripe.model.Event;
+import com.stripe.model.EventDataObjectDeserializer;
+import com.stripe.model.PaymentIntent;
+import com.stripe.model.StripeObject;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class PaymentEventParser {
+
+    public PaymentIntent parseEventToPaymentIntent(Event event) {
+        log.info("parse event to payment intent: preparation for payment event deserialization: event: {}", event);
+        EventDataObjectDeserializer dataObjectDeserializer = event.getDataObjectDeserializer();
+        StripeObject stripeObject = dataObjectDeserializer.getObject().orElseThrow();
+        log.info("parse event to payment intent: receiving stripe object: stripeObject: {}", stripeObject);
```

Comment:
StripeObject was not received, it was retrieved, wasn't it?
Update the log info